types/src/dkg/real_dkg/rounding/mod.rs

```diff
@@ -15,40 +16,46 @@ pub fn total_weight_lower_bound(validator_stakes: &Vec<u64>) -> usize {
     validator_stakes.len()
 }
 
+/// Find the smallest `stake_per_weight` that guarantees a valid rounding.
+/// Output the resulting total weight.
 pub fn total_weight_upper_bound(
     validator_stakes: &Vec<u64>,
-    reconstruct_threshold_in_stake_ratio: U64F64,
+    mut reconstruct_threshold_in_stake_ratio: U64F64,
     secrecy_threshold_in_stake_ratio: U64F64,
 ) -> usize {
-    assert!(reconstruct_threshold_in_stake_ratio > secrecy_threshold_in_stake_ratio);
-    let bound_1 = ((U64F64::from_num(1)
-        / (reconstruct_threshold_in_stake_ratio - secrecy_threshold_in_stake_ratio))
-        + U64F64::from_num(1))
-    .to_num::<u64>()
-        * (validator_stakes.len() as u64);
-
-    let stake_sum = validator_stakes.iter().sum::<u64>();
-    let stake_min = *validator_stakes.iter().min().unwrap();
-    let bound_2 = stake_sum / max(1, stake_min) + 1;
-
-    max(bound_1 as usize, bound_2 as usize)
+    reconstruct_threshold_in_stake_ratio = max(
+        reconstruct_threshold_in_stake_ratio,
+        secrecy_threshold_in_stake_ratio + U64F64::DELTA,
+    );
+    let n = U64F64::from_num(validator_stakes.len());
+    (n / U64F64::from_num(2)
```

Comment:
the formula looks very different now, it's probably also simpler to have a comment for the formula

Reply:
it's a bunch of statistics, thresholds, and the final weight vec.

Reply:
fixed by commenting `total_weight_upper_bound()`